frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -0,0 +1,625 @@
+
+        it("should show most recently visited dashboard if the question is in a public collection", async () => {
+          await setup({
+            mostRecentlyViewedDashboard: dashboardInPersonalSubcollection,
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([
+            ROOT_COLLECTION,
+            PERSONAL_COLLECTION,
+            PERSONAL_SUBCOLLECTION,
+          ]);
+
+          // dashboard item
+          expect(
+            await screen.findByText(dashboardInPersonalSubcollection.name),
+          ).toBeInTheDocument();
+        });
+
+        it("should show most recently visited dashboard if the question is in a personal collection", async () => {
+          await setup({
+            card: CARD_IN_PERSONAL_COLLECTION,
+            mostRecentlyViewedDashboard: dashboardInPersonalSubcollection,
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([
+            ROOT_COLLECTION,
+            PERSONAL_COLLECTION,
+            PERSONAL_SUBCOLLECTION,
+          ]);
+
+          // dashboard item
+          expect(
+            await screen.findByText(dashboardInPersonalSubcollection.name),
+          ).toBeInTheDocument();
+        });
+      });
+    });
+
+    describe("when user didn't visit any dashboard during last 24hrs", () => {
+      it("should render root collection without preselection", async () => {
+        await setup();
+
+        // breadcrumbs show root collection only
+        expect(screen.getByTestId("item-picker-header")).toHaveTextContent(
+          ROOT_COLLECTION.name,
+        );
+      });
+
+      describe("question is in a public collection", () => {
+        describe('"Create a new dashboard" option', () => {
+          it('should render "Create a new dashboard" option when opening the root collection (public collection)', async () => {
+            await setup();
+
+            expect(
+              screen.getByRole("heading", {
+                name: "Create a new dashboard",
+              }),
+            ).toBeInTheDocument();
+          });
+
+          it('should render "Create a new dashboard" option when opening public subcollections', async () => {
+            await setup();
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: COLLECTION.name,
+              }),
+            );
+
+            expect(
+              screen.getByRole("heading", {
+                name: "Create a new dashboard",
+              }),
+            ).toBeInTheDocument();
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: SUBCOLLECTION.name,
+              }),
+            );
+
+            expect(
+              screen.getByRole("heading", {
+                name: "Create a new dashboard",
+              }),
+            ).toBeInTheDocument();
+          });
+
+          it('should render "Create a new dashboard" option when opening personal subcollections', async () => {
```

Comment:
test Query builder > Newly created question > #4 from [the epic](https://github.com/metabase/metabase/issues/34733)